src/libraries/Microsoft.Extensions.Hosting.WindowsServices/src/WindowsServiceLifetimeHostBuilderExtensions.cs

```diff
@@ -49,13 +50,19 @@ public static IHostBuilder UseWindowsService(this IHostBuilder hostBuilder, Acti
                 hostBuilder.UseContentRoot(AppContext.BaseDirectory);
                 hostBuilder.ConfigureLogging((hostingContext, logging) =>
                 {
+                    Debug.Assert(RuntimeInformation.IsOSPlatform(OSPlatform.Windows));
```

Comment:
> Debug.Assert(RuntimeInformation.IsOSPlatform(OSPlatform.Windows)); [](http://example.com/codeflow?start=19&length=67)

what is the value of these asserts? is it possible at all something can go wrong here when we already checking with WindowsServiceHelpers.IsWindowsService()?

Reply:
It is only for suppressing the platform compat analyzer warning, apparently [by design](https://github.com/dotnet/roslyn-analyzers/issues/4282#issuecomment-702994171) the flow analysis could not evaluate lambda expressions in this case

Reply:
Correct - this is a workaround for the analyzer limitation.